packages/error-reporting/test/unit/google-apis/auth-client.js

```diff
@@ -41,24 +41,35 @@ function verifyReportedMessage(errToReturn, expectedMessage) {
   var logger = {
     error: function(text) {
       message += text;
-    }
+    },
+    info: function() {}
   };
-  var config = new Configuration({ ignoreEnvironmentCheck: true }, logger);
+  var config = new Configuration(config, logger);
   new RequestHandler(config, logger);
   assert.strictEqual(message, expectedMessage);
 }
-
 describe('RequestHandler', function() {
+  it('should not request OAuth2 token if key is provided', function() {
+    var config = {
+      ignoreEnvironmentCheck: true,
+      key: 'key'
+    };
+    var message = 'Made OAuth2 Token Request';
+    verifyReportedMessage(config, new Error(message), '');
```

Comment:
I think you would also want to verify that a call to `info` was done that had the text `"API key provided; skipping OAuth2 token request."` provided to verify that the user is provided with the correct message.